assembl/static2/js/app/components/debate/common/ideasLevel.jsx

```diff
@@ -0,0 +1,408 @@
+    this.moveAllChildren = this.moveAllChildren.bind(this);
+    this.onScrollRightClick = this.onScrollRightClick.bind(this);
+    this.onScrollLeftClick = this.onScrollLeftClick.bind(this);
+    this.scrollToIdeaIndexIfNecessary = this.scrollToIdeaIndexIfNecessary.bind(this);
+    this.refMe = this.refMe.bind(this);
+    this.refRow = this.refRow.bind(this);
+    this.updateWidth = this.updateWidth.bind(this);
+  }
+
+  componentDidMount() {
+    this.initializeConstants();
+  }
+
+  onSeeSubIdeasClick(ideaId, index, step = 0) {
+    const wasInline = this.state.isInline || this.getOption('isAnimatingTowardsInline');
+
+    if (!wasInline) {
+      const goNextStep = () => {
+        this.onSeeSubIdeasClick(ideaId, index, step + 1);
+      };
+      goNextStep.bind(this);
+
+      if (!step) {
+        /*
+        Animation step 0:
+          - Animate ideas top margins from various values to 0
+          - Set row width to its current width, in anticipation of future animation
+          - Set ideas left position to their current left position (0), in anticipation of future scroll animation
+          - Set ideas width to current width, in anticipation of future resize animation
+        */
+        const row = this.getRow();
+        row.style.width = `${row.clientWidth}px`;
+        row.style.transition = 'width 0.5s';
+        row.childNodes.forEach((el) => {
+          const el2 = el;
+          el2.style.width = `${el2.clientWidth}px`;
+          el2.style.marginTop = '0';
+          el2.style.left = '0';
+          el2.style.flexShrink = '0';
+        });
+
+        setTimeout(goNextStep, 500);
+        return;
+      } else if (step === 1) {
+        /*
+        Animation step 1:
+          - Set ideas positioning to a single row (set row display to inline-flex)
+          - Set necessary values consequently to inline-flex, for animation consistency
+        */
+        this.initializeConstants();
+
+        const row = this.getRow();
+
+        row.style.overflowX = 'hidden';
+        row.style.display = 'inline-flex';
+
+        setTimeout(goNextStep, 50);
+        return;
+      } else if (step === 2) {
+        /*
+        Animation step 2:
+          - Animate row width from full page width to custom value
+          - Animate ideas width from initial width to carousel-adapted width
+        */
+
+        this.setOption('isAnimatingTowardsInline', true);
+        this.updateWidth();
+
+        const row = this.getRow();
+        const thematicWidthPx = `${this.getOption('thematicWidth')}px`;
+        row.childNodes.forEach((el) => {
+          const el2 = el;
+          el2.style.width = thematicWidthPx;
+        });
+
+        this.scrollToIdeaIndexIfNecessary(index);
+
+        const f = () => {
+          this.onSeeSubIdeasClick(ideaId, index, step + 1);
+          this.forceUpdate();
+        };
+        f.bind(this);
+
+        setTimeout(f, 500);
```

Comment:
Yes, that's because animation is in progress. As animation is in several steps, we wait until the animation of a step is finished before executing the next step.